submissions/vakulinina/js-post-oop/index.js

```diff
@@ -0,0 +1,56 @@
+class Inhabitant {
+   constructor(species, name, gender, legs, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.legs = legs;
+      this.friends = [];
+   }
+
+   makeFriends(...friends) {
+      this.friends = [...friends];
+   }
+
+   getFriendsList(friends) {
+      return friends.length > 0 ? friends.map(friend => friend.name).join(', ') : 'no friends';
+   }
+
+   introduce() {
+      return [this.species, this.name, this.gender, this.legs, this.saying, this.getFriendsList(this.friends)].join('; ');
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying) {
+      super('human', name, gender, 2, saying);
+      this.hands = 2;
+   }
+
+   introduce() {
+      return [this.species, this.name, this.gender, this.legs, this.hands, this.saying, this.getFriendsList(this.friends)].join('; ');
+   }
```

Comment:
I see why you refer all the props here.
What would you change in order to have a scalable and maintainable implementation if allowed to expose properties in arbitrary order?
Imagine we need to establish dozens of species, so most likely we do not want every species `introduce` list all props as any change in the list of properties required for various species will require code changes in multiple classes and will quickly become a nightmare.

Reply:
Thank you, I was struggling to come up with a neater solution, but failed 😄 
Hope new one is ok.